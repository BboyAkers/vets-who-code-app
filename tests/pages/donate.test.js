import React from 'react'
import Donate from '../../src/pages/donate'

describe('<Donate />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Donate />)
  })

  test('should set modalIsOpen to true when button is clicked', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper.state('modalIsOpen')).toBe(true)
  })

  test('should set modalIsOpen to false when clicking outside modal', () => {
    wrapper.setState({ modalIsOpen: true })
    wrapper.instance().closeModal()
    wrapper.instance().forceUpdate()
    expect(wrapper.state('modalIsOpen')).toBe(false)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
