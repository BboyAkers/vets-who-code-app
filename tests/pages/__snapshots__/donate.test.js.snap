// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Donate /> should render correctly 1`] = `
<TemplateWrapper>
  <header
    className="inner-header overlay grey text-center slim-bg "
    style={
      Object {
        "backgroundImage": "url(image-mock)",
        "backgroundPositionY": "bottom",
      }
    }
  >
    <div
      className="overlay-01"
    />
    <div
      className="container"
    >
      <h2
        className="text-center text-uppercase"
      >
        Donate
      </h2>
      <div
        className="breadcrumb"
      >
        <Link
          to="/"
        >
          Home
        </Link>
        <span>
          /
        </span>
        <Link
          className="page-active"
          to="/donate"
        >
          Donate
        </Link>
      </div>
    </div>
  </header>
  <section
    className="cause_single section bg-default single pad-regular"
    id="cause_singe"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="cause_image_holder"
          >
            <img
              alt="cause"
              className="img-responsive"
              src="image-mock"
            />
          </div>
          <h4
            className="cause_title"
          >
            Help Us Continue To Train More Veterans
          </h4>
          <br />
          <div
            className="cause_section_content"
          >
            <p>
              Since 2014 we have been successfully training veterans in web development. From holding speaking engagements with industry leaders, to curriculum and teaching, to setting up tours and meetings with top tech companies, we have been on the front lines getting our troops the training and tools that they need to earn the opportunities that the tech industry offers. TO do that we need support from people like you, who not only support veterans but who understand the value of hard work and pushing yourself to obtain a goal. Please continue your support by giving a small donation so that we can continue to train more veterans.
            </p>
            <blockquote>
              <p>
                <img
                  alt="John Garcia"
                  src="image-mock"
                />
                <br />
                "VWC helped me gain the technical knowledge I needed in order to get the attention of employers. The guidance, support and experience I had going through the program continues to help me in my role as a full time web developer."
                <br />
                 - John Garcia, USAF | Web Developer, Hearst Media
              </p>
            </blockquote>
          </div>
        </div>
      </div>
    </div>
  </section>
  <DonateModal
    closeModal={[Function]}
    modalIsOpen={false}
    openModal={[Function]}
  />
  <section
    className="call-to-action-small"
    id="call-to-action-small"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-8 text-center"
        >
          <h3>
            Help Us Teach More Veterans How To Code
          </h3>
        </div>
        <div
          className="col-md-4"
        >
          <div
            className="button-align text-center"
          >
             
            <button
              className="dbox-donation-button"
              onClick={[Function]}
              style={
                Object {
                  "MozBorderRadius": 2,
                  "WebkitBorderRadius": 2,
                  "background": "#c5203e url(https://d1iczxrky3cnb2.cloudfront.net/white_logo.png) no-repeat 55px center",
                  "borderRadius": 2,
                  "boxShadow": "0 1px 0 0 #06152C",
                  "color": "#fff",
                  "display": "inline-block",
                  "fontFamily": "Verdana,sans-serif",
                  "fontSize": 16,
                  "padding": "15px 50px 15px 90px",
                  "textDecoration": "none",
                  "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
                }
              }
            >
              Donate
            </button>
          </div>
        </div>
      </div>
    </div>
     
  </section>
</TemplateWrapper>
`;
